scripts/add_extra_components.py

```diff
@@ -106,15 +106,17 @@ def attach_stores(n, costs):
                carrier='H2',
                e_nom_extendable=True,
                e_cyclic=True,
-               capital_cost=costs.at["hydrogen storage", "capital_cost"])
+               capital_cost=costs.at["hydrogen storage", "capital_cost"],
+               marginal_cost=costs.at["H2", "marginal_cost"])
```

Comment:
Shouldn't this be "hydrogen storage" rather than "H2"? To my understanding "H2" is for the `StorageUnit` and would equal the marginal costs for the fuel cell (dispatch only). Also, the marginal cost of a `Store` component refers to the dispatch. What would be the case for a marginal cost on the store rather than the links (electrolysis / fuel cell)?

Reply:
Great points. 
- Yes, we can make it consistent and name it "hydrogen storage" rather than "H2". 
- Maybe also worth changing the carrier name from "H2" to "H2 storage" as the carrier for the fuel cell are named "H2 fuel cell"/"H2 electrolyser"?
- Question. Are the marginal cost at `Store` not penalising the storage level? That's how I imagined its function or what you mean its related to the dispatch?
- I am not sure yet where this makes to add marginal cost to sense to add marginal cost to `Store` but giving the opportunity to do so might be useful. 
- I am writing up an unintended storage cycling (USC) study & how marginal cost can avoid that. In short, you can add marginal cost to either electrolyser, storage, discharger or your close to zero marginal cost generators (renewables) to avoid USC. Solver accuracy plays also a big role.

Reply:
> Yes, we can make it consistent and name it "hydrogen storage" rather than "H2".

Great

> Maybe also worth changing the carrier name from "H2" to "H2 storage" as the carrier for the fuel cell are named "H2 fuel cell"/"H2 electrolyser"?

Would prefer to keep it as H2, because the `Store` stores the carrier, whereas electrolysers and fuel cell convert. But I will keep it in mind also considering increased integration with PyPSA-Eur-Sec.

> Question. Are the marginal cost at Store not penalising the storage level? That's how I imagined its function or what you mean its related to the dispatch?

No, it penalises when the store is emptied (dispatched). See e.g. https://github.com/PyPSA/PyPSA/blob/d2c1a0805ccf23acfbabf4cb3bf3a931115b3718/pypsa/opf.py#L1183-L1185 for the pyomo version.

> I am not sure yet where this makes to add marginal cost to sense to add marginal cost to Store but giving the opportunity to do so might be useful.

In this case, I think it'd be better to leave it out then. This functionality is easy to add once needed.

> I am writing up an unintended storage cycling (USC) study & how marginal cost can avoid that. In short, you can add marginal cost to either electrolyser, storage, discharger or your close to zero marginal cost generators (renewables) to avoid USC. Solver accuracy plays also a big role.

Yes, that's what we have been doing. Quite an easy fix.